Consolidate Link props

carkom opened the pull request:
Consolidates AnchorLink, ButtonLink and LinkButton to use existing props (Link and Button - with type).

MatissJanis commented on .eslintrc.js:
```diff
@@ -146,10 +146,10 @@ module.exports = {
         selector: 'JSXOpeningElement[name.name="a"]',
         message:
-          'Using <a> is discouraged, please use <LinkButton> or <ExternalLink> instead.',
+          'Using <a> is discouraged, please use <Button type={link}> or <ExternalLink> instead.',
```
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
```suggestion
          'Using <a> is discouraged, please use <Button type="link"> or <ExternalLink> instead.',
```

MatissJanis commented on .eslintrc.js:
```diff
@@ -146,10 +146,10 @@ module.exports = {
       },
       {
-        // forbid <a> in favor of <LinkButton> or <ExternalLink>
+        // forbid <a> in favor of <Button type={'link'}> or <ExternalLink>
```
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
```suggestion
        // forbid <a> in favor of <Button type="link"> or <ExternalLink>
```

MatissJanis reviewed (approved):
This is definitely an improvement!

What still confuses me though is: `<Link type="button">` and `<Button type="link">`. We should eventually get rid of one of these.

carkom commented:
> This is definitely an improvement!
> 
> What still confuses me though is: `<Link type="button">` and `<Button type="link">`. We should eventually get rid of one of these.

<Link type="button"> - is an element that looks like a button but uses a "to" prop for navigating inside the app (eg. "Back" buttons).

`<Button type="link">` - does not really need to be a button because it doesn't actually look like a button. It looks like a link but uses an "onClick" prop to call a function rather than navigate to another page.

All the other "link" types we use need a "to" prop sent to them in order to not error out. I guess the second option above is a work around for making something look like a link but function like a button.

carkom commented:
I've created a new "ClickLink" in the Link file to be used instead of the `<Button type="link">`. 

Thoughts?

carkom commented:
@MatissJanis - all our link options are under one roof. I moved over external links as well as removing the "link" type in `<Button>`. I renamed all the options so they are more logical and easier to follow.

MatissJanis reviewed (approved):
This is great!

MatissJanis commented:
Closes #1414
